src/pubtools/sign/signers/cosignsigner.py

```diff
@@ -275,10 +274,16 @@ def container_sign(self: CosignSigner, operation: ContainerSignOperation) -> Sig
                 args.append(ref_digest)
                 ref_args.append(args)
 
-        for args in ref_args:
-            outputs.append(run_command(common_args + args, env=env_vars, tries=self.retries))
+        run_command_args = [
+            FData(
+                args=[common_args + args],
+                kwargs={"env": env_vars, "tries": self.retries},
+            )
+            for args in ref_args
+        ]
+        ret = run_in_parallel(run_command, run_command_args)
```

Comment:
Running in parallel for the same reference may cause conflict (multiple requests writing to the same signature, not all are written successfully). Could you ensure requests of each reference are run in line?

Reply:
Fixing it by grouping cosign command by reference, then running groups simultaneously and executing sequentially in each group.